### PR TITLE
Drop support for python 3.7

### PR DESCRIPTION
**Pull request by qduk**

### CODE REVIEW

**Review by jeffkala — COMMENTED** [review]
Just making sure we want to drop 3.7 before eol which is late June

**Comment by qduk** [comment]
> Just making sure we want to drop 3.7 before eol which is late June

Should be ready whenever we do want to merge.

**Comment by itdependsnetworks** [comment]
I would wait until EOL.

**Comment by jeffkala** [comment]
@qduk EOL for py37 passed now. We can probably get this one pushed in.

**Review by jeffkala — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ include = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = "^3.8"
```
Comment (jeffkala) [inline pyproject.toml:28]: remove classifiers for 3.7
classifiers = [
    .
    .
    "Programming Language :: Python :: 3.7",
]

File: tasks.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def tests(context, local=INVOKE_LOCAL):
         context (obj): Used to run specific commands
         local (bool): Define as `True` to execute locally
     """
-    black(context, local)
+    # black(context, local)
```
Comment (jeffkala) [inline tasks.py:252]: can we clean this back up?

Comment (qduk) [reply tasks.py]: Yes.

File: pyproject.toml (hunk 1 above)
Comment (qduk) [reply pyproject.toml:27]: Updated

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,6 +17,8 @@ jobs:
         uses: "actions/checkout@v2"
       - name: "Setup environment"
         uses: "networktocode/gh-action-setup-poetry-environment@v2"
+        with:
```
Comment (itdependsnetworks) [inline .github/workflows/ci.yml:20]: why do we have to have these?

Comment (qduk) [reply .github/workflows/ci.yml:20]: It's not needed. I believe we discussed having the tests run on 3.11 for the speed improvements. Want me to remove them?

Comment (itdependsnetworks) [reply .github/workflows/ci.yml:20]: What version would be it be otherwise? it's not clear to me how this is being used.

Comment (qduk) [reply .github/workflows/ci.yml:20]: The action [sets up](https://github.com/networktocode/gh-action-setup-poetry-environment/blob/main/action.yml) the poetry environment. It defaults to version 3.9.

File: netutils/config/parser.py [hunk 4]
```diff
@@ -52,7 +49,6 @@ class BaseSpaceConfigParser(BaseConfigParser):
     """Base parser class for config syntax that demarcates using spaces/indentation."""
 
     # pylint: disable=abstract-method
```
Comment (glennmatthews) [inline netutils/config/parser.py:51]: remove, per comment?

File: pyproject.toml [hunk 5]
```diff
@@ -13,7 +13,6 @@ classifiers = [
     "Intended Audience :: Developers",
     "Development Status :: 4 - Beta",
```
Comment (glennmatthews) [inline pyproject.toml:14]: Possibly could/should update this at some point. :-)

File: pyproject.toml [hunk 6]
```diff
@@ -13,7 +13,6 @@ classifiers = [
...
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
```
Comment (glennmatthews) [inline pyproject.toml:18]: Perhaps out of scope, but what's needed to enable this for python 3.11?
```suggestion
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
```

**Review by glennmatthews — APPROVED** [review]

**Comment by itdependsnetworks** [comment]
addressed in #317

**Pull request closed without merging**